addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -139,11 +139,9 @@ def _timesheet_get_portal_domain(self):
             # Then, he is internal user, and we take the domain for this current user
             return self.env['ir.rule']._compute_domain(self._name)
         return ['&',
-                    '|', '|', '|',
+                    '|',
                     ('task_id.project_id.message_partner_ids', 'child_of', [self.env.user.partner_id.commercial_partner_id.id]),
                     ('task_id.message_partner_ids', 'child_of', [self.env.user.partner_id.commercial_partner_id.id]),
```

Comment:
Vérifier si il ne faut pas ajouter : ('task_id.message_partner_ids', 'child_of', [self.env.user.partner_id.id]),

Reply:
By consulting history, it seems to be aligned with the preceding behavior. So I suggest to keep it as it is.